gen/generate.go

```diff
@@ -144,21 +145,95 @@ func SequenceAttrProto(a *parse.Attribute, wSeq, wAttr io.Writer) error {
 
 		// Write out the sequence protobuf message.
 		i := 1
-		for _, s := range a.SubAttributes {
-			fmt.Fprintf(wSeq, "\t%s %s = %d;\n", s.Keyword, getFieldName(s.Name), i)
+		attrs := getSortedAttributes(a.SubAttributes)
+		for _, attr := range attrs {
+			fmt.Fprintf(wSeq, "\t%s %s = %d;\n", attr.Keyword, getFieldName(attr.Name), i)
 			i++
 		}
 		fmt.Fprint(wSeq, "} \n\n")
 
 		// Recursively call for children.
-		for _, s := range a.SubAttributes {
+		for _, s := range attrs {
 			SequenceAttrProto(s, wSeq, wAttr)
 		}
 
 	}
 	return nil
 }
 
+func getSortedKeys(m map[parse.TagKey]*parse.Attribute) []string {
```

Comment:
these can alternatively be defined on a custom type in `parse` for (`map[parse.TagKey]*parse.Attribute`) this way users of the package have access to them as well.